c2corg_ui/static/js/documentediting.js

```diff
@@ -314,56 +266,25 @@ app.DocumentEditingController.prototype.submitForm = function(isValid) {
       'message': message,
       'document': data
     };
-    this.http_.put(this.buildUrl_('update'), data, config).then(
-        this.successSave_.bind(this),
-        this.errorSave_.bind(this)
+    this.api_.updateDocument(this.module_, this.id_, data).then(function() {
+      window.location.href = app.utils.buildDocumentUrl(
+        this.module_, this.id_, this.lang_);
+    }.bind(this)
     );
   } else {
     // creating a new document
     this.lang_ = data['locales'][0]['lang'];
-    this.http_.post(this.buildUrl_('create'), data, config).then(
-        this.successSave_.bind(this),
-        this.errorSave_.bind(this)
-    );
+    this.api_.createDocument(this.module_, data).then(function(response) {
+      this.id_ = response['data']['document_id'];
+      window.location.href = app.utils.buildDocumentUrl(
+        this.module_, this.id_, this.lang_);
+    }.bind(this));
   }
   // TODO: show a loading message until doc saving is done
```

Comment:
This line (TODO) can be removed since a loading indicator has been added last week.